### ISSUE TITLE
Skip to content action not working properly with Voiceover

### ISSUE THREAD

**Issue by DBD324** [mi6/ic-design-system#616]
## Summary of the bug

When using the skip to content link, or skip to page link, the focus position is moved, but not the virtual cursor position. 

This means when navigating through a page using the virtual cursor, the skip to content and skip to page links don't do anything. The virtual cursor just moves to the element after the skip link instead of into the content that its supposed to skip to. 

Seems to be a voiceover issue when tested on chrome. When testing with NVDA the functionality works and the virtual cursor is repositioned to the correct place along with the focus position. 

## 🪜 How to reproduce

Tell us the steps to reproduce the problem:

1. Go to page: Buttons page
2. Turn on voiceover
3. Tab to 'skip to main content' link and press enter. 
4. See the focus position change to the correct element. 
5. Use the virtual cursor to move to next element. 
6. Virtual cursor position remains at top of page and goes to element after skip to main content link. 

(same for skip to page content)

## 🖥 📱 Device

- Type: Laptop
- Device: Mac
- OS version: 
- Browser version: Chrome
- Voiceover screen reader

## 🧐 Expected behavior

When using the skip buttons, both the focus position and the virtual cursor should be moved to the content. 
After triggering the skip buttons, then the virtual cursor should move through the content from the position that the skip button directs you. 

Use BBC News Homepage as exemplar. The Skip To Content function there works as expected

## 📝 Acceptance Criteria

_Given_ that I am using voiceover
_When_  triggering the skip to main content link, and then continuing to navigate with the virtual cursor
_Then_ the first element in the main content should be navigated to. 

_Given_ that I am using voiceover
_When_  triggering the skip to page content link, and then continuing to navigate with the virtual cursor
_Then_ the first element in the page content should be navigated to. 

## Additional info

It might be that we're choosing the wrong element as the target for our skip to content functionality, and that's confusing the virtual cursor. Maybe try setting a different element as the target (and make note of the element that BBC uses as the target) 

Once this ticket is closed, it might be valuable to update the Accessibility guidance for our Back To Top button, to advise developers on best practice for setting the 'target' prop.

**Comment by GCHQ-Developer-847** [comment]
## Research

### Skip to main content link

Looking at the "Skip to _main_ content" link, I have found that there are a few different approaches that have been taken on other websites:
- **Option 1:** Set the target element to `<main>`, e.g. on [GOV.UK](https://www.gov.uk/).
- **Option 2:** Add a visually hidden `<h1>` element as the first element of the `<main>` and set that as the target element, e.g. [BBC News](https://www.bbc.co.uk/news).
- **Option 3:** Add an`<a>` element with no href as the first element of the `<main>`and set that as the target element, e.g. [W3C](https://www.w3.org/TR/WCAG20-TECHS/intro.html).

In addition to this, it seems that the key to getting the virtual cursor (rather than just keyboard focus) to move to the target element of the skip to content link is to apply `tabindex="-1"` to it. A few sources say to add `tabindex="-1"` but I cannot find much information actually explaining  _why_ this works. It does work though; it seems that all examples of skip to content links online that work for the virtual cursor have this.


#### Thoughts / findings for each option

Option 1
- This is the same sort of set-up we have on the ICDS website at the moment, and when I add the `tabindex="-1"`, it does improve things because the VoiceOver user can actually hear that the focus has moved (previously, it didn't actually announce that anything had happened).
- It is fine on the home page but it seems to be a little problematic with the virtual cursor when on other pages which have a "Skip to _page_ content" link (as well as the "Skip to _main_ content" link). When I try VO + Right arrow (see [VoiceOver keyboard shortcuts](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts)) after I've pressed Enter on the "Skip to main content link", it just goes straight past the "Skip to page content" link as if it doesn't exist.


Option 2
- This option seemed quite good at first because it reads out a title to the screen reader user, so it's very clear to them what part of the page they are on.
- However, using an `<h1>` is not really an option because we already have an `<h1>` in the top navigation. The way BBC News do it is to not have an `<h1>` in their top navigation; the home link is just a `<span>`, but the `<h1>` in the ic-top-navigation is built-in, even if you're using a slot. (We could change this at some point though if we wanted to but it would likely be a breaking change).
- I tried using a `<p>` or a `<span>` instead but I feel like this makes the layout of the page a bit more confusing than if I could just use an `<h1>`, because these elements don't signify the start of a section like a heading element does. I could technically use an `<h2>` but this would require changing all the other heading elements across the website to ensure they are in the correct order with no duplicates.


Option 3
- This option doesn't have the virtual cursors that the first one has. All VoiceOver keyboard shortcuts for the virtual cursor pick up the "Skip to page content" link fine.
- It doesn't require any of the refactoring of the website's structure that Option 2 would.
- I was worried at first that the `<a>` would get picked up in VoiceOver's rotor (VO + U), but I have tested this and it doesn't.
- This is how [W3C](https://www.w3.org/TR/WCAG20-TECHS/intro.html) do it, and it seems to be quite a common method used to take the user to different sections on a page.

I decided to go with Option 3 as this was the most effective solution, and easiest to implement at this point without restructuring the ICDS website.

----------

### Skip to page content link

I tried just adding the `tabindex="-1"` to the existing target element for the "Skip to page content" link - which is the container `<div>` for the page contents. However, when you press Enter on the link, it just starts reading out all of the information on the page. 

Using the same approach as I chose for the "Skip to main content" link (with an `<a>` element) seems to work, but I had to give it a name using an aria-label, otherwise VO just reads out "group". I think not using an aria-label for the "Skip to main content" link is fine though because it says "group, main".
